### PR TITLE
feat(crons): Allow users to download attachments

### LINKED ISSUE #43657: Crons: Enable attachments to be sent with a check-in
## Problem
When users see an `Failed` check-in on the crons dashboard or get alerted on a `Failed` check-in they are not given any additional context such as an error message to help them debug what happened during cron execution.

## Overview
To help with debugging check-in failures and ensuring the overall health of their system we should allow users to send additional data with the status of the check-in. This could include data such as error messages or metrics like the number of items processed during job execution.

## Proposed Solution
Add a new `File` column into the `monitor.py` model which can hold arbitrary data from the user. We will try to mimic the functionality of [event attachments](https://docs.sentry.io/platforms/javascript/enriching-events/attachments/) where the user can specify a MIME type for the file. Then make changes to the API to support sending/writing this data into the new column. Additionally, we will have to ensure this data follows the user retention policy.

## Integration into CLI
For users who choose to wrap their commands with the CLI instead of calling the API directly we can offer another option onto `sentry-cli monitor run` which will tell us to listen to STDOUT or STDERR when running the user's command and send the output to the API when registering a check-in.

## Related Tasks
```[tasklist]
### Related Tasks
- [ ] https://github.com/getsentry/sentry/issues/44701
```

### PR DESCRIPTION
**Pull request by davidenwang**
Adds a GET to `organizations/<org_id>/monitors/<monitor_id>/checkins/<checkin_id>/attachment/` which triggers a download for the attachment stored with the specified checkin. Returns a 404 if no attachment is there.

### CODE REVIEW

File: tests/sentry/api/endpoints/test_monitor_checkin_attachment.py [hunk 1]
```diff
@@ -81,6 +83,21 @@ def test_upload_no_file(self):
         assert resp.status_code == 400
         assert resp.data["detail"] == "Missing uploaded file"
 
+    def test_download_no_file(self):
+        monitor = self._create_monitor()
+        checkin = MonitorCheckIn.objects.create(
```
Comment (davidenwang) [inline tests/sentry/api/endpoints/test_monitor_checkin_attachment.py:88]: will move this to factories.py in follow up PR

File: src/sentry/api/endpoints/monitor_checkin_attachment.py [hunk 2]
```diff
@@ -18,6 +19,23 @@ class MonitorCheckInAttachmentEndpoint(MonitorCheckInEndpoint):
     private = True
     authentication_classes = MonitorCheckInEndpoint.authentication_classes + (DSNAuthentication,)
 
+    def download(self, file_id):
+        file = File.objects.get(id=file_id)
+        fp = file.getfile()
+        response = FileResponse(
+            fp,
+            content_type=file.headers.get("Content-Type", "application/octet-stream"),
```
Comment (evanpurkhiser) [inline src/sentry/api/endpoints/monitor_checkin_attachment.py:27]: So we don't know the mime type yet?

Comment (davidenwang) [reply src/sentry/api/endpoints/monitor_checkin_attachment.py:27]: we do, just supplying this as a default

Comment (evanpurkhiser) [reply src/sentry/api/endpoints/monitor_checkin_attachment.py:27]: oh I see it now

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**